### ISSUE TITLE
ConnectionPool::Error (no connections are checked out)

### ISSUE THREAD

**Issue by retiolus** [mastodon/mastodon#20671]
### Steps to reproduce the problem

1. after updating from release candidate 3 to 4
2. `journalctl -u mastodon-web -f`
3. 
...

### Expected behaviour

Should work

### Actual behaviour

The pages must be reloaded constantly for the content to load (Error 500), see logs below

### Detailed description

```
Nov 14 18:37:16 kfem bundle[1462673]: [478895a0-2c6f-4bc2-a5ea-cec76c0121c5] method=POST path=/inbox format=html controller=ActivityPub::InboxesController action=create status=202 duration=21.54 view=0.00 db=1.87 key=https://mastodon.art/users/Ailantd#main-key
Nov 14 18:37:18 kfem bundle[1462675]: [275d08bb-579e-41e0-b403-7819d5275bf7] method=POST path=/inbox format=html controller=ActivityPub::InboxesController action=create status=202 duration=23.00 view=0.00 db=0.99 key=https://mastodon.nz/users/narofi6719#main-key
Nov 14 18:37:19 kfem bundle[1462675]: [2e7108ca-d144-4a2f-ada2-bc2269c0e53b] method=POST path=/inbox format=html controller=ActivityPub::InboxesController action=create status=202 duration=19.77 view=0.00 db=3.18 key=https://mastodon.cloud/users/mteodoro#main-key
Nov 14 18:37:20 kfem bundle[1462673]: [d8b79627-2db7-4ca4-8c72-643fb9eec7b6] method=POST path=/inbox format=html controller=ActivityPub::InboxesController action=create status=202 duration=27.81 view=0.00 db=1.70 key=https://fosstodon.org/users/MattTheQuick#main-key
Nov 14 18:37:20 kfem bundle[1462673]: [d8b79627-2db7-4ca4-8c72-643fb9eec7b6]
Nov 14 18:37:20 kfem bundle[1462673]: [d8b79627-2db7-4ca4-8c72-643fb9eec7b6] ConnectionPool::Error (no connections are checked out):
Nov 14 18:37:20 kfem bundle[1462673]: [d8b79627-2db7-4ca4-8c72-643fb9eec7b6]
Nov 14 18:37:20 kfem bundle[1462673]: [d8b79627-2db7-4ca4-8c72-643fb9eec7b6] lib/mastodon/rack_middleware.rb:22:in `clean_up_redis_socket!'
Nov 14 18:37:20 kfem bundle[1462673]: [d8b79627-2db7-4ca4-8c72-643fb9eec7b6] lib/mastodon/rack_middleware.rb:17:in `clean_up_sockets!'
Nov 14 18:37:20 kfem bundle[1462673]: [d8b79627-2db7-4ca4-8c72-643fb9eec7b6] lib/mastodon/rack_middleware.rb:11:in `call'
Nov 14 18:37:21 kfem bundle[1462673]: [82ffdd30-7fce-4f61-92f4-ceeaef361642] method=POST path=/inbox format=html controller=ActivityPub::InboxesController action=create status=202 duration=25.63 view=0.00 db=2.08 key=https://social.espeweb.net/users/elnacionalcat_bot#main-key
Nov 14 18:37:22 kfem bundle[1462675]: [264326b2-48fb-4374-9975-fe054894762c] method=POST path=/inbox format=html controller=ActivityPub::InboxesController action=create status=202 duration=28.26 view=0.00 db=2.10 key=https://mastodon.social/users/lolgop#main-key
Nov 14 18:37:26 kfem bundle[1462675]: [42bfa48b-ea9a-469e-be36-a48a027e5e67] method=GET path=/api/v1/statuses/109342890138465261/context format=html controller=Api::V1::StatusesController action=context status=200 duration=44.42 view=2.07
```

### Specifications

Mastodon v4.0.0rc4
Redis server v=5.0.7
node v16.16.0

**Comment by ineffyble** [comment]
Have you double-checked Redis settings and that Redis is online and accessible?

**Comment by retiolus** [comment]
I haven't changed the redis settings since the previous version, and it's online fine, plus the pages load, just half the time not

**Comment by retiolus** [comment]
This error seemed to be related to version 4.0.0rc3 (maybe), since I upgraded the server to 4.0.0rc4 and later I have not had this error.

**Comment by xpat** [comment]
I'm getting the same errors on one of my mastodon servers.  After today's update to version 4.0.0, having followed the upgrade guide, I can't get the server working again.  I've restarted all three mastodon-* and checked status.  They're all active. And then again... streaming changes from active to failed.

**Comment by turion** [comment]
I'm experiencing the same problem as @retiolus , with 4.0.2!

**Comment by turion** [comment]
@xpat how is your error related? It seems like a different one.

My error seems to be resolved by `systemctl restart mastodon-web.service`.

**Comment by xpat** [comment]
> @xpat how is your error related? It seems like a different one.
> 
> My error seems to be resolved by `systemctl restart mastodon-web.service`.

@turion I removed then reinstalled vendor/bundle and restarted all three processes and got it working that way.

**Comment by gnh1201** [comment]
I don't understand why this issue is closed, but I will attach an error that occurred in my system.

Node version: 18.12.1

```bash
$ NODE_ENV=production /usr/bin/node ./streaming
/home/mastodon/live/node_modules/jsdom/lib/jsdom/browser/parser/html.js:170
    after._pushedOnStackOfOpenElements?.();
                                       ^

SyntaxError: Unexpected token '.'
    at wrapSafe (internal/modules/cjs/loader.js:915:16)
    at Module._compile (internal/modules/cjs/loader.js:963:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/mastodon/live/node_modules/jsdom/lib/jsdom/browser/parser/index.js:4:20)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
```

**Comment by gnh1201** [comment]
I think I've found a solution. I noted that this error is `SyntaxError`. If a `SyntaxError` occurs in a package that has already been tested, it is possible that the runtime version is too old.

I used the `node --version` command to verify this. But my system returned me the wrong version information. A typical user would mistake it for using the appropriate node version.

The node version provided by the package manager provided by the operating system (e.g., apt or yum) is likely to be lower than 16. Nevertheless, my system lied that the node version it had was equal to or higher than 16, and the reason was not known until I checked the result of the `where node` command.

```bash
$ whereis node
node: /usr/bin/node /home/mastodon/.nvm/versions/node/v19.3.0/bin/node /usr/share/man/man1/node.1.gz
```

`/usr/bin/node` does not refer to the node of the NVM (or the user installed node).

So, If we change the systemd profile as below, the problem can be solved.

```diff
-ExecStart=/usr/bin/node ./streaming
+ExecStart=/home/mastodon/.nvm/versions/node/v19.3.0/bin/node ./streaming
```

And then. run `systemctl daemon-reload`
